Travis fails building PRs from other forks

Branches from my fork of SIRF that I want to PR into master fail with the following error:

```
Cloning into 'SIRF'...
Note: checking out 'e66070e1a3e6f8a00dd515dc7b43177207a58301'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

fatal: reference is not a tree: 7c3e5ea1fd3e19bba75c0b8ac905da02b80a91ae
CMake Error at /home/travis/build/CCPPETMR/SIRF-SuperBuild/builds/SIRF/tmp/SIRF-gitclone.cmake:75 (message):
  Failed to checkout tag: '7c3e5ea1fd3e19bba75c0b8ac905da02b80a91ae'
```

Until now I've simply pushed to the CCPPETMR branch, but this isn't possible for everyone (nor is it a long-term solution).

--- a/.travis.yml
+++ b/.travis.yml
@@ -113,8 +113,8 @@ addons:
 
 # cache C/C++/pip
 cache:
-  - ccache
-  - pip
+ - ccache
+ - pip
 
 before_install:
  # Set C and C++ compiler etc using trick from
@@ -224,6 +224,8 @@ before_install:
 
  - export PATH="$PWD:$PATH"
  - popd
+ # ensure a branch exists (for e.g. PRs)
+ - git symbolic-ref -q HEAD || git checkout -b local_$TRAVIS_COMMIT
  # Use Travis' currently checked-out SIRF commit ID to build.
  # Also no point re-downloading SIRF - just use local URL.
  # N.B.: don't put into build matrix to allow caching.

